examples/axual/main.tf

```diff
@@ -0,0 +1,467 @@
+}
+
+resource "axual_application_access_grant" "log_scraper_consume_from_support_in_dev" {
+  application = axual_application.log_scraper.id
+  stream = axual_stream.support.id
+  environment = axual_environment.development.id
+  access_type = "CONSUMER"
+  depends_on = [ axual_application_principal.log_scraper_in_dev_principal ]
+}
+
+resource "axual_application_access_grant" "dash_consume_from_logs_in_staging" {
+  application = axual_application.dev_dashboard.id
+  stream = axual_stream.logs.id
+  environment = axual_environment.staging.id
+  access_type = "CONSUMER"
+  depends_on = [ axual_application_principal.dev_dashboard_in_staging_principal ]
+}
+
+resource "axual_application_access_grant" "dash_consume_from_support_in_staging" {
+  application = axual_application.dev_dashboard.id
+  stream = axual_stream.support.id
+  environment = axual_environment.staging.id
+  access_type = "CONSUMER"
+  depends_on = [ axual_application_principal.dev_dashboard_in_staging_principal ]
+}
+
+resource "axual_application_access_grant" "scraper_produce_to_logs_in_staging" {
+  application = axual_application.log_scraper.id
+  stream = axual_stream.logs.id
+  environment = axual_environment.staging.id
+  access_type = "PRODUCER"
+  depends_on = [ axual_application_principal.log_scraper_in_staging_principal ]
+}
+
+resource "axual_application_access_grant" "dash_consume_from_logs_in_production" {
+  application = axual_application.dev_dashboard.id
+  stream = axual_stream.logs.id
+  environment = axual_environment.production.id
+  access_type = "CONSUMER"
+  depends_on = [ axual_application_principal.dev_dashboard_in_production_principal ]
+}
+
+resource "axual_application_access_grant" "dash_consume_from_support_in_production" {
+  application = axual_application.dev_dashboard.id
+  stream = axual_stream.support.id
+  environment = axual_environment.production.id
+  access_type = "CONSUMER"
+  depends_on = [ axual_application_principal.dev_dashboard_in_production_principal ]
+}
+
+resource "axual_application_access_grant" "scraper_produce_to_logs_in_production" {
+  application = axual_application.log_scraper.id
+  stream = axual_stream.logs.id
+  environment = axual_environment.production.id
+  access_type = "PRODUCER"
+  depends_on = [ axual_application_principal.log_scraper_in_production_principal ]
+}
+
+#
+# An APPLICATION_ACCESS_GRANT can be approved by creating an APPLICATION_ACCESS_GRANT_APPROVAL with
+# a reference to the APPLICATION_ACCESS_GRANT which needs to be approved, as can be seen
+# in the example below
+#
+# Reference: https://registry.terraform.io/providers/Axual/axual/latest/docs/resources/application_access_grant_approval
+#
+
+resource "axual_application_access_grant_approval" "dash_consume_logs_dev" {
+  application_access_grant = axual_application_access_grant.dash_consume_from_logs_in_dev.id
+  depends_on = [ axual_application_access_grant.dash_consume_from_logs_in_dev ]
```

Comment:
Can we remove these **depends_on** from **axual_application_access_grant_approval** and **axual_application_access_grant_rejection** because this depends_on currently has no effect, because Terraform implicitly already depends on **axual_application_access_grant** meaning that Terraform will not create **axual_application_access_grant_approval** before Terraform has gotten a 200 from API for creating **axual_application_access_grant**. https://developer.hashicorp.com/terraform/language/meta-arguments/depends_on So we can safely remove these depends_on statements.